ops/op2/dispatch_fast.rs

```diff
@@ -95,104 +103,162 @@ pub fn generate_dispatch_fast(
+    None => return Ok(None),
+    Some(rv) => rv,
   };
 
   let GeneratorState {
     fast_function,
     deno_core,
+    result,
+    opctx,
+    fast_api_callback_options,
+    needs_fast_api_callback_options,
+    needs_fast_opctx,
     ..
-  } = &generator_state;
+  } = generator_state;
 
-  let input_types = inputs.iter().map(|fv| fv.quote_type());
+  let handle_error = match signature.ret_val {
+    RetVal::Infallible(_) => quote!(),
+    RetVal::Result(_) => {
+      *needs_fast_api_callback_options = true;
+      *needs_fast_opctx = true;
+      inputs.push(FastValue::CallbackOptions);
+      quote! {
+        let #result = match #result {
+          Ok(#result) => #result,
+          Err(err) => {
+            unsafe { #opctx.unsafely_set_last_error_for_ops_only(err); }
+            #fast_api_callback_options.fallback = true;
+            return ::std::default::Default::default();
+          }
+        };
+      }
+    }
+  };
+
+  let input_types = inputs.iter().map(|fv| fv.quote_type()).collect::<Vec<_>>();
   let output_type = output.quote_ctype();
 
   let fast_definition = quote! {
     use #deno_core::v8::fast_api::Type;
     use #deno_core::v8::fast_api::CType;
     #deno_core::v8::fast_api::FastFunction::new(
-      &[ #( #input_types ),* ],
+      &[ Type::V8Value, #( #input_types ),* ],
       #output_type,
       Self::#fast_function as *const ::std::ffi::c_void
     )
   };
 
-  let output_type = output.quote_rust_type();
-  let names = &inputs
+  let output_type = output.quote_rust_type(deno_core);
+  let mut types = inputs
     .iter()
-    .enumerate()
-    .map(|(i, _)| format_ident!("arg{i}"))
+    .map(|rv| rv.quote_rust_type(deno_core))
     .collect::<Vec<_>>();
-  let types = inputs.iter().map(|rv| rv.quote_rust_type());
+
+  let call_args = names.clone();
+
+  let with_fast_api_callback_options = if *needs_fast_api_callback_options {
+    types.push(FastValue::CallbackOptions.quote_rust_type(deno_core));
+    names.push(fast_api_callback_options.clone());
+    quote! {
+      let #fast_api_callback_options = unsafe { &mut *#fast_api_callback_options };
+    }
+  } else {
+    quote!()
+  };
+  let with_opctx = if *needs_fast_opctx {
+    quote!(
+      let #opctx = unsafe {
+        &*(#deno_core::v8::Local::<v8::External>::cast(unsafe { #fast_api_callback_options.data.data }).value()
+            as *const #deno_core::_ops::OpCtx)
+      };
+    )
+  } else {
+    quote!()
+  };
 
   let fast_fn = quote!(
     fn #fast_function(
       _: #deno_core::v8::Local<#deno_core::v8::Object>,
       #( #names: #types, )*
     ) -> #output_type {
-      #(
-        let #names = #names as _;
-      )*
-      Self::call(#(#names),*)
+      #with_fast_api_callback_options
+      #with_opctx
+      let #result = Self::call(#(#call_args as _),*);
+      #handle_error
+      #result
     }
   );
 
   Ok(Some((fast_definition, fast_fn)))
 }
+
+fn map_fast_arg(arg: &Arg) -> Result<Option<FastValue>, V8MappingError> {
```

Comment:
Nitpick, but can we be more specific in naming? Ie. instead of using `fast` can we use `v8_fast_call` everywhere? It's a bit longer but less ambiguous than just "fast"